docs(agw): Adding Ubuntu AGW migration guide (#7929)
magma/magma: docs/readmes/lte/migration_guide.md

---
id: migration_guide

> title: AGW migration guide

Reviewer: ```suggestion
title: Migrate AGWs to Ubuntu
```

hide_title: true
---

> # Access Gateway migration guide

Reviewer: ```suggestion
# Migrate AGWs from Debian to Ubuntu
```

> NOTE: Only Ubuntu-based Access Gateways are supported starting from Magma release 1.6, Debian support is deprecated.

> For a smooth transition from a Debian to Ubuntu access gateway, it is advised to backup local configs including `RootCA`, `control_proxy.yml`, `Hardware ID` and `Challenge Key` files. It is also recommended to backup any static IPs or routes provisioned on the existing access gateway.

Reviewer: ```suggestion
For a smooth transition from a Debian to Ubuntu access gateway, back up local configs including

- Gateway's hardware ID
- Gateway's long-term challenge key (`gw_challenge.key`)
- Gateway's control_proxy config (`control_proxy.yml`)
- Root CA certificate (`rootCA.pem`)
- Any static IPs or routes provisioned directly on the existing gateway
```

> ### Before migration

Reviewer: ```suggestion
## Before migration
```

> #### Copy the contents from the following file locations to a local backup storage:

Reviewer: ```suggestion
Copy the contents from the following file locations to a local backup storage
```

> - `Control Proxy`
> > */var/opt/magma/configs/control_proxy.yml*

Reviewer: For this and the ones below

Also, put them in the same order as the previous section

```suggestion
- Control proxy config: `/var/opt/magma/configs/control_proxy.yml`
```

- `Root CA`
> */var/opt/magma/tmp/certs/rootCA.pem*

- `Hardware ID`
> */etc/snowflake*

- `Challenge Key`
> */var/opt/magma/certs/gw_challenge.key*

> After copying these, gather the output of `show_gateway_info.py` as reference and follow the [Ubuntu Install Guide](https://docs.magmacore.org/docs/lte/deploy_install_ubuntu)

Reviewer 2: the deployment link is changed on master, can you change deploy_install_ubuntu  to deploy_install.

> ### After migration

Reviewer: ```suggestion
## After migration
```

- Once Magma is installed on the Access Gateway, copy the contents from local backup to configure the new access gateway

> - Restart `magmad` services to enable the new configs:
> > *service magma@magmad restart*

Reviewer: ```suggestion
- Restart `magmad` services to enable the new configs: `service magma@magmad restart`
```

- The output of `show_gateway_info.py` should match the one captured before migration

> - Verify access gateway is checked-in to the existing orchestrator and configs are being streamed successfully:
> > *journalctl -fu magma@magmad*

Reviewer: ```suggestion
- Verify access gateway is checked-in to the existing Orchestrator and configs are being streamed successfully: `journalctl -fu magma@magmad`
```
